### ISSUE TITLE
LoadStrategy.SELECT_IN broken within transactions.

### ISSUE THREAD

**Issue by mollerhoj** [mikro-orm/mikro-orm#4443]
**Describe the bug**

I have 2 entities, Dashboard and Widget:

```
@Entity()
export class Dashboard extends BaseEntity<Dashboard, 'id'> {
  @PrimaryKey({ type: 'number', autoincrement: true })
  id!: number;

  @Property({ type: 'string' })
  name!: string;

  @OneToMany('Widget', 'dashboard', { cascade: [Cascade.ALL], strategy: LoadStrategy.JOINED })
  widgets = new Collection<Widget>(this);
}

@Entity()
export class Widget extends BaseEntity<Widget, 'id'> {
  @PrimaryKey({ type: 'number', autoincrement: true })
  id!: number;

  @Property({ type: 'string' })
  name!: string;

  @ManyToOne('Dashboard')
  dashboard!: Dashboard;
}
```

A test:
```
  it("populate", async ({ em }) => {
    em.begin()
  
    let dashboard = em.create(Dashboard, { name: "J" });
    let w1 = em.create(Widget, { name: "J", position: 0, dashboard: dashboard });
    let w2 = em.create(Widget, { name: "J", position: 1, dashboard: dashboard });
    em.persistAndFlush([dashboard, w1, w2]);

    let loadedDashboard = await em.findOneOrFail(Dashboard, { name: "J" }, { populate: ["widgets"] });
    expect(loadedDashboard.widgets.length).toBe(2);

    await em.rollback();
  });
```

**Stack trace**
```
...
```

**To Reproduce**
Change the LoadStrategy to SELECT_IN.

Observe that the test also passes if run outside a transaction.

**Expected behavior**
I expect to test to pass.

**Additional context**
For some reason, the `select * from widget` queries are not being fired when with a transaction context.

**Versions**

| Dependency | Version |
| - | - |
| node | 19.9.0 |
| typescript | 5.0.4 |
| mikro-orm | 5.7.8 |

**Comment by SpencerKaiser** [comment]
This stems from the identity map not having knowledge of the collection's contents. [Repro with fix (added `.set`)](https://stackblitz.com/edit/mikro-orm-v6-strict-partial-loading-zv9hcd?file=basic.test.ts).

**Comment by B4nan** [comment]
The problem here is that you (or at least the repro from Spencer) is using target ES2022, so has `useDefineForClassFields` enabled, and that means there is no propagation on entity prototype. This is partially a wontfix (with `useDefineForClassFields` enabled there is no way to modify the prototype of those properties) and partially solved by #4434 in v6 - that PR makes sure propagation works when you use `em.create` (entity constructors will still fail).
